httpbp/headers.go

```diff
@@ -0,0 +1,182 @@
+func GetHeader(ctx context.Context, key HeaderContextKey) (header string) {
+	if h, ok := ctx.Value(headerContextKey(key)).(string); ok {
+		header = h
+	}
+	return
+}
+
+// Headers is an interface to collect all of the HTTP headers for a particular
+// baseplate resource (spans and edge contexts) into a struct that provides an
+// easy way to convert them into HTTP headers.
+//
+// This interface exists so we can avoid having to do runtime checks on maps to
+// ensure that they have the right keys set when we are trying to sign or verify
+// a set of HTTP headers.
+type Headers interface {
+	// AsMap returns the Headers struct as a map of header keys to header
+	// values.
+	AsMap() map[string]string
+}
+
+// EdgeContextHeaders implements the Headers interface for HTTP EdgeContext
+// headers.
+type EdgeContextHeaders struct {
+	EdgeRequest string
+}
+
+// NewEdgeContextHeaders returns a new EdgeContextHeaders object from the given
+// HTTP headers.
+func NewEdgeContextHeaders(h http.Header) EdgeContextHeaders {
+	return EdgeContextHeaders{
+		EdgeRequest: h.Get(EdgeContextHeader),
+	}
+}
+
+// AsMap returns the EdgeContextHeaders as a map of header keys to header
+// values.
+func (s EdgeContextHeaders) AsMap() map[string]string {
+	return map[string]string{
+		EdgeContextHeader: s.EdgeRequest,
+	}
+}
+
+// SpanHeaders implements the Headers interface for HTTP Span headers.
+type SpanHeaders struct {
+	TraceID  string
+	ParentID string
+	SpanID   string
+	Flags    string
+	Sampled  string
+}
+
+// NewSpanHeaders returns a new SpanHeaders object from the given HTTP headers.
+func NewSpanHeaders(h http.Header) SpanHeaders {
+	return SpanHeaders{
+		TraceID:  h.Get(TraceIDHeader),
+		ParentID: h.Get(ParentIDHeader),
+		SpanID:   h.Get(SpanIDHeader),
+		Flags:    h.Get(SpanFlagsHeader),
+		Sampled:  h.Get(SpanSampledHeader),
+	}
+}
+
+// AsMap returns the SpanHeaders as a map of header keys to header values.
+func (s SpanHeaders) AsMap() map[string]string {
+	return map[string]string{
+		TraceIDHeader:     s.TraceID,
+		ParentIDHeader:    s.ParentID,
+		SpanIDHeader:      s.SpanID,
+		SpanFlagsHeader:   s.Flags,
+		SpanSampledHeader: s.Sampled,
+	}
+}
+
+var (
+	_ Headers = EdgeContextHeaders{}
+	_ Headers = SpanHeaders{}
+)
+
+// InjectBaseplateContextFromHTTP takes baseplate HTTP headers from the request,
+// verifies that it should trust the headers using the provided
+// HeaderTrustHandler, and attaches the trusted headers to the context.
+//
+// These headers can be retrieved using httpbp.GetHeader.
+//
+// This method does not implement the go-kit http.RequestFunc interface, if you
+// want to use this with go-kit, use PopulateBaseplateRequestContext to
+// return a function that wraps InjectBaseplateContextFromHTTP with the
+// HeaderTrustHandler and does implement the http.RequestFunc interface.
+func InjectBaseplateContextFromHTTP(ctx context.Context, t HeaderTrustHandler, r *http.Request) context.Context {
```

Comment:
🔕 These function names are getting long, I'm wondering if we could get them shorter, for example `ContextFromHTTP` for this one, `PopulateRequestContext` for the next one.